### PR TITLE
nss: add nss-config (required by xulrunner etc.)

### PR DESCRIPTION
**Pull request by l-hedgehog**
Similar to #9705, some configure scripts (firefox, xulrunner etc.)
check for nss via nss-config, which is not available upstream.

GNU/Linux distros ship their own variants of nss-config, e.g.
Debian: http://anonscm.debian.org/cgit/pkg-mozilla/nss.git/tree/debian/nss-config.in
Fedora: http://pkgs.fedoraproject.org/cgit/nss.git/tree/nss-config.in
Gentoo: http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/dev-libs/nss/files/nss-3.17.1-gentoo-fixups.patch?view=markup

The one in this pr is mostly a wrapper around pkg-config.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Sorry but this is way more logic I'm happy keeping and maintaining in a formula. This is something that should be submitted and worked on upstream.

**Comment by l-hedgehog** [comment]
@mikemcquaid, would you consider a simpler version, just enough for building firefox/xulrunner?

**Comment by mikemcquaid** [comment]
@l-hedgehog if it's like 1-2 lines: maybe. I still think it would be a better approach to work with upstream so there's a single version everyone can use rather than every package manager having their own one.

**Comment by l-hedgehog** [comment]
@mikemcquaid, I understand, I'll try to come up with a simpler one, and I can always use my own fork.

As working with upstream, this has been discussed since like ten years ago, similar to nss.pc. Sigh...

**Comment by l-hedgehog** [comment]
Hi @mikemcquaid, would you take a look at this version of nss-config? Thanks!

I also triggered a rebuild of xulrunner 33.0 on top of it to make sure it's good enough for firefox/xulrunner (--with-system-nss is the correct option to build xulrunner with system nss, not --with-nss-prefix, and sorry for pushing a few bad changes as I'm on Yosemite).

**Comment by mikemcquaid** [comment]
> Hi @mikemcquaid, would you take a look at this version of nss-config? Thanks!

Looks sufficiently minimal but would still love to see something submitted upstream too if that's possible (or at least a link to an upstream bug report).

**Comment by l-hedgehog** [comment]
https://bugzil.la/255408
https://bugzil.la/530672

**Comment by mikemcquaid** [comment]
This needs rebased on `master` and if you could link those issues in the formula file itself at the relevant place that'd be great!

**Comment by l-hedgehog** [comment]
In case a force push won't trigger any notification: this was rebased onto a recent master (later than nss 3.17.4), with a link to relevant bug as comment. I also tweaked this nss-config script a bit.

**Comment by mikemcquaid** [comment]
Yeh, it doesn't trigger notifications unfortunately, thanks for letting me know.

File: Library/Formula/nss.rb [hunk 1]
```diff
@@ -67,8 +65,18 @@ def install
     system "#{bin}/certutil", "-L", "-d", pwd
   end
 
+  # A very minimal nss-config for configuring firefox etc. with this nss,
+  # see https://bugzil.la/530672 for the progress of upstream inclusion.
+  def config_file; <<-EOS.undent
+    #!/bin/sh
+    for opt; do :; done
+    case "$opt" in --version) opt="--modversion";; --cflags|--libs) ;; *) exit 1;; esac
```
Comment (mikemcquaid) [inline Library/Formula/nss.rb]: Let's split this onto multiple lines, it's hard to follow as-is.

File: Library/Formula/nss.rb [hunk 2]
```diff
@@ -67,8 +65,18 @@ def install
     system "#{bin}/certutil", "-L", "-d", pwd
   end
 
+  # A very minimal nss-config for configuring firefox etc. with this nss,
+  # see https://bugzil.la/530672 for the progress of upstream inclusion.
+  def config_file; <<-EOS.undent
+    #!/bin/sh
+    for opt; do :; done
```
Comment (mikemcquaid) [inline Library/Formula/nss.rb]: What's this do? My bash-fu isn't that strong.

Comment (l-hedgehog) [inline Library/Formula/nss.rb]: With a full featured nss-config, it's possible to change its output like this:
````
~ » nss-config --prefix=/alt/prefix --cflags
-I/alt/prefix/include
````
The ones supported by this minimal version are guaranteed to be the last argument when configuring firefox/xulrunner, so I just ignore any others.

Comment (mikemcquaid) [inline Library/Formula/nss.rb]: Ah so you're just getting the last argument here, OK.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by l-hedgehog** [comment]
Thank you!
